pulser/simulation/simulation.py

```diff
@@ -139,6 +136,12 @@ def set_config(self, cfg: SimConfig) -> None:
         """
         if not isinstance(cfg, SimConfig):
             raise ValueError(f"Object {cfg} is not a valid `SimConfig`.")
+        not_supported = set(cfg.noise) - SUPPORTED_NOISE[self._interaction]
```

Comment:
What if instead of creating the new variable `not_supported` every time, we check `set(cfg.noise).issubset(SUPPORTED_NOISE[self._interaction])`, and compute the difference only when the error is raised?

Reply:
I think it's a trade-off here: when you fail, you already have the noises you don't support, so you save on computation time there. In the other case, indeed it is a bit more expensive, though I would venture guessing it's negligible. Considering that this is not called often, is it worth the potential optimization?